src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -208,15 +220,17 @@ namespace OpenMS
       // only add to string if not a fixed modification
       if (std::find(fixed_modifications.begin(), fixed_modifications.end(), cterm_mod_name) == fixed_modifications.end())
       {
-        double nominal_mass = mod.getDiffMonoMass(); // just get input mass
-        if (!mod.isUserDefined()) nominal_mass += Residue::getInternalToCTerm().getMonoWeight();
+        double nominal_mass = Residue::getInternalToCTerm().getMonoWeight() + mod.getDiffMonoMass();
```

Comment:
there are a few places where this computation is performed, but the result might be discarded immediately after if `mass_delta == true`. For readibility and performance in Debug mode, can you switch to ternary op or if-else. 
~~I dont think its a performance problem in Release~~  (scratch that... no compiler I checked will get rid of the computation... ouch https://godbolt.org/z/xds49Y). Even more reason do change it.

Reply:
e.g.
```
double nominal_mass = mod.getDiffMonoMass();
if (mass_delta) nominal_mass =+ Residue::getInternalToCTerm().getMonoWeight() ;
```

Reply:
> if (mass_delta) nominal_mass =+ Residue::getInternalToCTerm().getMonoWeight() ;

I guess you mean if (!mass_delta)